internal/rest/dequeue_nodes.go

```diff
@@ -423,8 +423,18 @@ func (rest *RestOperations) ExecuteRestAndPopulateCache(rest_ops []*utils.RestOp
 								utils.AviLog.Warnf("key: %s, msg: retry count exhausted, skipping", key)
 							}
 						} else {
-							utils.AviLog.Warnf("key: %s, msg: Avi model not set", key)
-							retry = true
+							utils.AviLog.Warnf("key: %s, msg: Avi model not set, possibly a DELETE call", key)
+							aviError, ok := rest_ops[i].Err.(session.AviError)
+							// If it's 404, don't retry
+							if ok {
+								statuscode := aviError.HttpStatusCode
+								if statuscode != 404 {
+									rest.PublishKeyToSlowRetryLayer(publishKey, aviclient, key)
+									return
+								} else {
+									rest.AviVsCacheDel(rest_ops[i], aviObjKey, key)
```

Comment:
Can this come here during a static route deletion? @parimanur / @monotosh-avi

Reply:
As discussed we dont set the model to nil for vrfcontext.